apps/aecore/src/aec_sync.erl

```diff
@@ -257,125 +277,106 @@ fetch_headers(PeerUri, GHash, Acc) ->
 fetch_headers_1(HdrHash, Hdr, PeerUri, GHash, Acc) ->
     case aec_chain:get_header_by_hash(HdrHash) of
         {ok, _} ->
-            case aec_chain:get_block_by_hash(HdrHash) of
-                {ok, _} ->
-                    lager:debug("we have the top block - done", []),
-                    %% We're done already!
-                    ok;
-                {error, _} ->
-                    lager:debug("fetching top block (~p)", [HdrHash]),
-                    Acc1 = fetch_block(HdrHash, PeerUri, Acc),
-                    fetch_next(aec_headers:prev_hash(Hdr),
-                               PeerUri, GHash, Acc1)
-            end;
+            %% we have a continuous header chain
+            headers_fetched(PeerUri, Acc);
         {error, _} ->
             lager:debug("we don't have the top block header,"
-                        " fetching block (~p)", [HdrHash]),
+                        " fetching block (~p)", [pp(HdrHash)]),
             Acc1 = fetch_block(HdrHash, PeerUri, Acc),
```

Comment:
Comment out of scope for this PR, as related to sync rather than filling in of missing blocks:
It looks like that, when syncing, the node does not download and validate the header chain first and *then* download the blocks, but rather go straight for the each block.
@gorillainduction Is this what you meant in your comment https://www.pivotaltracker.com/story/show/152484212/comments/182108262 ?